### PR TITLE
[KEYCLOAK-14335] - Not initializing entity associations and removing bi-directional ones

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java [hunk 1]
```diff
@@ -1245,8 +1249,8 @@ public ClientModel getMasterAdminClient() {
 
     @Override
     public void setMasterAdminClient(ClientModel client) {
-        ClientEntity appEntity = client !=null ? em.getReference(ClientEntity.class, client.getId()) : null;
-        realm.setMasterAdminClient(appEntity);
+        String appEntityId = client !=null ? em.getReference(ClientEntity.class, client.getId()).getId() : null;
+        realm.setMasterAdminClient(appEntityId);
         em.flush();
```
Comment (stianst) [inline model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java:1254]:
```suggestion
        realm.setMasterAdminClient(client.getId());
```

I don't think we need to lookup the client here, nor do we need an em.flush here.

**Review by stianst — COMMENTED** [review]
Looks good to me, but think we can improve the setMasterAdminClient method a bit. See inline comment

**Review by hmlnarik — DISMISSED** [review]
I like the changes here. Hence approving.

Still, I was thinking of for a long time - is there really a need to use `ArrayList` instead of `LinkedList`? If we used indexed access to the returned items, then fine, but I don't think we ever do in the code, the returned lists are rather iterated through. If you agree with this reasoning, I believe this PR is the best candidate to change this.

**Comment by stianst** [comment]
> I like the changes here. Hence approving.
> 
> Still, I was thinking of for a long time - is there really a need to use `ArrayList` instead of `LinkedList`? If we used indexed access to the returned items, then fine, but I don't think we ever do in the code, the returned lists are rather iterated through. If you agree with this reasoning, I believe this PR is the best candidate to change this.

Good point - as we return them as Collection there's not really any way we can use them as an ArrayList right, so there's not much point in ArrayList.

**Comment by pedroigor** [comment]
@hmlnarik Good point considering that when the entity is not yet initialized we are probably going to perform adds/removes on those collections and LinkedList is a better option for this than ArrayList. 

Will change that and see how it goes.

**Review by stianst — APPROVED** [review]

**Pull request merged**